wagtailmodelchoosers/views.py

```diff
@@ -26,30 +27,35 @@ def get_params(self):
         return self.request.parser_context.get('kwargs')
 
     def do_search(self, cls, queryset):
-        search = self.request.query_params.get('search', None)
-        if not search:
+        """
+        Comma-delimited search for neighborhood w/ city, state
+        """
+        search_value = self.request.query_params.get('search', None)
+        if not search_value:
             return queryset
 
-        queries = []
+        query_list = map(lambda str: str.strip(), string.split(search_value, ','))
         if hasattr(cls, 'SEARCH_FIELDS'):
-            queries = [Q(**{search_field: search}) for search_field in cls.SEARCH_FIELDS]
-        else:
-            for field in cls._meta.get_fields():
-                if isinstance(field, CharField):
-                    kwargs = {}
-                    param_name = '%s__icontains' % field.name
-                    kwargs[param_name] = search
-                    queries.append(Q(**kwargs))
+            query = Q()
+            for index, search_field in enumerate(cls.SEARCH_FIELDS):
+                query &= Q(**{search_field: query_list[index]})
+            return queryset.filter(query)
+
+        queries = []
+        search_fields = [field.name for field in cls._meta.get_fields() if isinstance(field, CharField)]  # NOQA
+        for field in search_fields:
+            kwargs = {}
+            param_name = '%s__icontains' % field.name
```

Comment:
`search_fields` looks like it's already a list of field names, shouldn't the param_name here be just `field`?